ACM-37751 - NetworkPolicy md (#529)
stolostron/insights-metrics: docs/NETWORK_POLICIES.md

  namespace-wide policy would inadvertently restrict traffic for pods that aren't part of
  Insights.
- **`policyTypes: [Ingress]` only.** OVN-Kubernetes (the default CNI on OpenShift) handles
  `kubernetes.default.svc` ClusterIP traffic through the OVN service load balancer *before*
  NetworkPolicy evaluation, so no egress rule type can match kube-API traffic. Applying an
  Egress policyType would silently block insights-metrics from reaching the Kubernetes API.
- **Well-known namespace labels.** Ingress rules that reference OpenShift system namespaces
  use the `kubernetes.io/metadata.name` label, which the API server automatically stamps on
  every namespace (Kubernetes 1.21+). This avoids relying on custom labels that may not exist
  in every cluster.

## Component network flows

### insights-metrics

| Direction | Peer | Port | Rationale |
|---|---|---|---|
| Ingress | `openshift-monitoring` namespace | 8443/TCP | Prometheus (`prometheus-k8s`) scrapes metrics through the kube-rbac-proxy sidecar, which terminates TLS on port 8443 and proxies to the metrics container on `127.0.0.1:8383`. The `ServiceMonitor` resource configures a 60-second scrape interval. |

> | Egress | *(not restricted — Ingress-only policy)* | — | insights-metrics requires egress to the Kubernetes API to watch cluster resources for metrics collection (PolicyReports, ManagedClusters, etc.) and to poll the APIServer TLS security profile. OVN-Kubernetes handles `kubernetes.default.svc` ClusterIP traffic via the OVN service load balancer before NetworkPolicy evaluation, so no egress rule can match kube-API traffic. Applying an Egress policyType would silently block the metrics collector from reaching the Kubernetes API. |

Reviewer: 🤖 I'm an AI agent

Agreeing with CodeRabbit here — "not restricted by this NetworkPolicy" is more precise than "not restricted — Ingress-only policy", since other NetworkPolicies selecting the same pods could still restrict egress.

```suggestion
| Egress | *(not restricted by this NetworkPolicy)* | — | insights-metrics requires egress to the Kubernetes API to watch cluster resources for metrics collection (PolicyReports, ManagedClusters, etc.) and to poll the APIServer TLS security profile. OVN-Kubernetes handles `kubernetes.default.svc` ClusterIP traffic via the OVN service load balancer before NetworkPolicy evaluation, so no egress rule can match kube-API traffic. Applying an Egress policyType would silently block the metrics collector from reaching the Kubernetes API. |
```
